float8_experimental/float8_ops.py

```diff
@@ -140,3 +141,29 @@ def autocast_to_copy(aten_op, args, kwargs=None):
     return Float8Tensor(
         args[0]._data, args[0]._scale, kwargs["dtype"], args[0]._emulate
     )
+
+
+@implements([c10d_functional.all_gather_into_tensor.default])
+def allgather_fp8(aten_op, args, kwargs=None):
+    """
+    override funcol with FP8 handling
+    """
+    fp8_input = args[0]
+    assert isinstance(fp8_input, Float8Tensor)
```

Comment:
maybe a helpful message, you would be surprised what happends during torch.compile lol